services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java

```diff
@@ -297,46 +299,82 @@ private JaxrsSAML2BindingBuilder buildLogoutBinding(KeycloakSession session, Use
+            sp.setEntityId(getEntityId(uriInfo, realm));
+            sp.setNameIDPolicyFormat(getConfig().getNameIDPolicyFormat());
+            sp.setDisplayName(getConfig().getDisplayName());
+            sp.setDescription(getConfig().getDescription());
+            sp.setInformationURL(getConfig().getInformationURL());
+            sp.setPrivacyStatementURL(getConfig().getPrivacyStatementURL());
+            sp.setLogo(getConfig().getLogo());
+            sp.setLogoWidth(getConfig().getLogoWidth());
+            sp.setLogoHeight(getConfig().getLogoHeight());
+            sp.setGeantDataProtectionCode(getConfig().isGeantDataProtectionCode());
+            sp.setResearchAndScholarship(getConfig().isResearchAndScholarship());
+            sp.setRegistrationAuthority(getConfig().getRegistrationAuthority());
+            sp.setRegistrationPolicy(getConfig().getRegistrationPolicy());
+            if (getConfig().getOrganizationName() != null && getConfig().getOrganizationDisplayName() != null
+                && getConfig().getOrganizationURL() != null) {
+                sp.setOrganizationName(getConfig().getOrganizationName());
+                sp.setOrganizationDisplayName(getConfig().getOrganizationDisplayName());
+                sp.setOrganizationURL(getConfig().getOrganizationURL());
+            }
+            if (getConfig().getContactType() != null) {
+                sp.setContactType(ContactTypeType.valueOf(getConfig().getContactType()));
+                sp.setContactCompany(getConfig().getContactCompany());
+                sp.setContactGivenName(getConfig().getContactGivenName());
+                sp.setContactSurname(getConfig().getContactSurname());
+                sp.setContactEmailAddresses(Arrays.asList(getConfig().getContactEmailAddress().split(",")));
+                sp.setContactTelephoneNumbers(Arrays.asList(getConfig().getContactTelephoneNumber().split(",")));
             }
 
-            URI endpoint = uriInfo.getBaseUriBuilder()
-                    .path("realms").path(realm.getName())
-                    .path("broker")
-                    .path(getConfig().getAlias())
-                    .path("endpoint")
-                    .build();
-
-
-            boolean wantAuthnRequestsSigned = getConfig().isWantAuthnRequestsSigned();
-            boolean wantAssertionsSigned = getConfig().isWantAssertionsSigned();
-            boolean wantAssertionsEncrypted = getConfig().isWantAssertionsEncrypted();
-            String entityId = getEntityId(uriInfo, realm);
-            String nameIDPolicyFormat = getConfig().getNameIDPolicyFormat();
-
-            List<Element> signingKeys = new ArrayList<Element>();
-            List<Element> encryptionKeys = new ArrayList<Element>();
+            for (IdentityProviderMapperModel idpModel : realm.getIdentityProviderMappersByAliasAndMapper(getConfig().getAlias(),
+                "saml-user-attribute-idp-mapper")) {
```

Comment:
Use constants in this whole block, e.g.
```suggestion
                UserAttributeMapper.PROVIDER_ID)) {
```